src/constructs/iam/roles/fastly-kinesis-log.ts

```diff
@@ -0,0 +1,45 @@
+import { AccountPrincipal } from "aws-cdk-lib/aws-iam";
+import { FASTLY_AWS_ACCOUNT_ID } from "../../../constants/fastly-aws-account-id";
+import type { GuStack } from "../../core";
+import { GuFastlyCustomerIdParameter } from "../../core";
+import type { GuKinesisStream } from "../../kinesis";
+import { GuKinesisPutRecordsPolicy } from "../policies/kinesis-put-records";
+import { GuRole } from "./roles";
+
+export interface GuFastlyKinesisLogRoleProps {
+  /**
+   * The Kinesis stream into which Fastly will put records
+   */
+  stream: GuKinesisStream;
+  /**
+   * Name of the IAM role
+   */
+  roleName?: string;
+}
+
+/**
+ * A construct to create an IAM Role for Fastly to assume in order to write to a
+ * specific Kinesis stream.
+ *
+ * In order to use this construct, an SSM parameter named `/account/external/fastly/customer.id`
+ * needs to exist in the AWS account's parameter store, and the value should be
+ * the Guardian's Fastly customer id.
+ *
+ */
+export class GuFastlyKinesisLogRole extends GuRole {
```

Comment:
Nit: As this is a sibling of the `GuFastlyLogsIamRole` should this be one level up in the `constructs/iam` folder, or is the other role in the wrong place?

Reply:
I thought about this when I created the files and concluded that the other one (`GuFastlyLogsIamRole`) is possibly in the wrong place.
Since this PR adds a new `AWS::IAM::Role` I think the logical place for this construct to live is the `iam/roles` folder, in the same way that `AWS::IAM::Policy` constructs live in the `iam/policies` folder.